marklogic-client-api/src/main/java/com/marklogic/client/impl/RowManagerImpl.java

```diff
@@ -285,6 +285,30 @@ public RawSPARQLSelectPlanImpl newRawSPARQLSelectPlan(TextWriteHandle handle) {
     return handle.get();
   }
 
+  @Override
+  public <T extends JSONReadHandle> T columnInfo(Plan plan, T resultsHandle) {
+    if (resultsHandle == null) {
+      throw new IllegalArgumentException("Must specify a handle to generate a view for the plan");
+    }
+
+    PlanBuilderBaseImpl.RequestPlan requestPlan = checkPlan(plan);
+    AbstractWriteHandle astHandle = requestPlan.getHandle();
+
+    RequestParameters params = new RequestParameters();
+    params.add("output",     "columnInfo");
+
+    return services.postResource(requestLogger, "rows", null, params, astHandle, resultsHandle);
+  }
+  @Override
+  public <T> T columnInfoAs(Plan plan, Class<T> as) {
+    ContentHandle<T> handle = handleFor(as);
```

Comment:
Before casting to a JSONReadHandle, would it be appropriate to check JSONReadHandle.class.isInstance(handle) and throw an error if, for instance, the class is not an XMLDomHandle?

Reply:
Thanks for the suggestion. I'll add the check here.